datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -617,12 +619,24 @@ protected void handlePollRecordsException(Exception e) throws Exception {
       _logger.warn("Poll fail with exception", e);
       throw e;
     } else {
-      _logger.warn(
-          String.format("Poll threw an exception. Sleeping for %d seconds and repoll from consumer, poll attempt: %d",
-              _retrySleepDuration.getSeconds(), _pollAttempts.intValue()), e);
-    }
-    if (!_shutdown) {
-      Thread.sleep(_retrySleepDuration.toMillis());
+      _logger.warn("Failed to poll. repoll from consumer, poll attempt={}", _pollAttempts.intValue(), e);
+      // add partition into pause list on topic not authz for access
+      if (e instanceof TopicAuthorizationException) {
+        TopicAuthorizationException tae = (TopicAuthorizationException) e;
+        final Set<String> unauthorizedTopics = tae.unauthorizedTopics();
+        _logger.warn("Not authorized to access, they will be added to auto-pause set, topics={}", unauthorizedTopics);
+        for (String topic: unauthorizedTopics) {
+          List<PartitionInfo> partitionInfos = _consumer.partitionsFor(topic);
```

Comment:
what is the desired behavior if partitionsFor throws an exception in this scenario ?

Reply:
I don't see it throwing exception. However, `KafkaConsumer::partitionFor()` method description says there could be exception, interesting. In this case, I think we should just catch and ignore it, so the behavior would be same as what we have on prod version right now.